In [3]:
#Loading necessary libraries
library(tidyverse)
library(tidymodels)

In [14]:
#Loading csv files
players <- read_csv('data/players.csv')
sessions <- read_csv('data/sessions.csv')
#slice(players, 1:30)
#summary(players)
slice(sessions, 1:5)
summary(sessions)


Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          

# 1. Data Description

## players csv 

Rows: 139 
Columns: 7

**Summary Statistics**

| Variable | Type | Description | Notes and Key Stats |
|----------|------|-------------| --------------------|
| experience| character | categorizes players between "Beginner", "Amateur", "Regular", "Pro", and "Veteran" |  5.00 different classification labels.|
| subscribe | logical | indicates whether a player is subscribed ("TRUE") or not ("FALSE") to the newsletter | Variable can only be "TRUE" or "FALSE". |
|hashedEmail | character | a hashed email of the email address used by the player(s) | N/A
| played_hours | double | indicates the amount of time in hours of game played by players | Variable has a mean = 5.85 and max = 223.10. The variable in the dataset also reports values within one decimal place.|
|name | character | indicates the name of the player(s) | All observations start with a capital letter to denote a person's name. |
| gender | character | indicates the preferred gender of players | Can be classified between "Male", "Female", "Prefer not to say", "Non-Binary", "Two-Spirited", or "Other". |
|Age | double | indicates the age of players | The variable reports values within a whole number. Also has 2 N/As in the dataset |

## sessions csv

Rows: 1535
Columns: 5

**Summary Statistics** 

| Variable | Type | Description | Notes and Key Stats |
|----------|------|-------------| --------------------|
|hashedEmail | character | a hashed email of the email address used by the player(s) | N/A |
| start_time | character | indicates the start time of a player's session by date and 24-hour time format | The date is in dd/mm/yyyy format. |
| end_time | character | indicates the end time of a player's session by date and 24-hour time format | The date is in dd/mm/yyyy format. |
| original_start_time | double | indicates the start time of a player by UNIX time | The dataset conforms with scientific notation to represent the numbers. Variable mean = 1.719e+12 and max = 1.727e+12. |
| original_end_time | double | indicates the end time of a player by UNIX time | The dataset conforms with scientific notation to represent the numbers. Variable mean = 1.719e+12 and max = 1.727e+12 and has 2 N/As in the dataset.|


# 2. Question

**Can we predict whether a player will subscribe to the newsletter using experience and age?**